state/upgrades.go

```diff
@@ -1884,3 +1885,54 @@ func UpdateInheritedControllerConfig(pool *StatePool) error {
 	}
 	return nil
 }
+
+// EnsureDefaultModificationStatus ensures that there is a modification status
+// document for every machine in the statuses.
+func EnsureDefaultModificationStatus(pool *StatePool) error {
+	st := pool.SystemState()
+	db := st.db()
+
+	machineCol, machineCloser := db.GetRawCollection(machinesC)
+	defer machineCloser()
+	// Find all machines that are alive.
+	machineIter := machineCol.Find(bson.D{{"life", life.Alive}}).Iter()
+	defer machineIter.Close()
+
```

Comment:
you really should check the returned error on Iter.Close. it is the only thing that tells u if the db went away or the query was invalid.